cl/opinion_page/types.py

```diff
@@ -0,0 +1,80 @@
+from dataclasses import dataclass
+from typing import List
+
+from django.urls import reverse
+
+from cl.search.models import OpinionCluster, OpinionsCitedByRECAPDocument
+
+
+@dataclass
+class ViewAuthority:
+    caption: str
+    count: int
+    url: str
+
+    @classmethod
+    def from_cluster_authority(cls, auth: OpinionCluster, query_string: str):
+        opinion_url = f"{auth.get_absolute_url()}?{query_string}"
+        citation_count = auth.citation_depth
+        opinion_caption = auth.caption
+        return cls(
+            caption=opinion_caption, count=citation_count, url=opinion_url
+        )
+
+    @classmethod
+    def from_recap_cit_record(
+        cls, record: OpinionsCitedByRECAPDocument, query_string: str
+    ):
+        opinion_caption = record.cited_opinion.cluster.caption
+        citation_count = record.depth
+        opinion_url = (
+            f"{record.cited_opinion.cluster.get_absolute_url()}?{query_string}"
+        )
+        return cls(
+            caption=opinion_caption, count=citation_count, url=opinion_url
+        )
+
+
+@dataclass
+class AuthoritiesContext:
+    total_authorities_count: int
+    top_authorities: List[ViewAuthority]
+    view_all_url: str
+
+    @classmethod
+    def from_opinion_cluster(
```

Comment:
I wonder if you could merge this and the method below by sniffing the object type passed in. Did you play with that?

Reply:
@mlissner current version pushed now has an attempt of this, with initial version commented out.

Reply:
@mlissner nudge

Reply:
Sorry, I'm leaving this for @ERosendo to review, but he has a few higher priority items on his list first. We're really bogged down here at FLP, but we're digging out. I'm delegating more dev work to others, and we are just really behind on Elastic, which is causing everything else to suffer. 

Anyway, that's TMI, perhaps, but I expect Eduardo will have a minute for this soon.

Reply:
Sounds good